framework/src/actions/SetupMeshAction.C

```diff
@@ -258,7 +258,12 @@ SetupMeshAction::act()
   else if (_current_task == "set_mesh_base")
   {
     if (!_app.masterMesh())
-      _mesh->setMeshBase(_mesh->buildMeshBaseObject());
+    {
+      if (_awh.hasActions("add_mesh_generator"))
+        _mesh->setMeshBase(_app.getMeshGeneratorMesh());
+      else
+        _mesh->setMeshBase(_mesh->buildMeshBaseObject());
```

Comment:
This will create a conflict with #14077, but it is easy to fix: just need to add the guards. For now, just for the else branch. When I start using MeshGenerators in the component system, I might need to do something about if-branch, but may be not - it is hard to predict that...